main.go

```diff
@@ -329,15 +330,22 @@ func sanityCheck(args *arguments) exitCode {
 		}
 
 		var minMajor, minMinor uint32
-		switch runtime.GOARCH {
-		case "amd64":
-			minMajor, minMinor = 4, 19
-		case "arm64":
-			// Older ARM64 kernel versions have broken bpf_probe_read.
-			// https://github.com/torvalds/linux/commit/6ae08ae3dea2cfa03dd3665a3c8475c2d429ef47
-			minMajor, minMinor = 5, 5
-		default:
-			return failure("Unsupported architecture: %s", runtime.GOARCH)
+		if args.debugBpf {
+			log.Infof("The DEBUG version of the BPF program can significantly increase the " +
+				"number of compiled BPF instructions, requiring at least kernel version 5.8 " +
+				"to ensure that BPF verifier is happy")
+			minMajor, minMinor = 5, 8
```

Comment:
https://github.com/torvalds/linux/commit/c04c0d2b968ac45d6ef020316808ef6c82325a82 raised the number of allowed eBPF instructions and was released with Linux kernel 5.2. Which commit landed with 5.8, that the Linux kernel version should be pushed to 5.8 to be able to run eBPF programs that are compiled with debug options?

Reply:
I found this page through a search engine (https://lore.kernel.org/bpf/7ba1fa1f-be6-1fa2-1877-12f7b707b65@gmail.com/T/#m6a8744aa014a5b5f0360b9d9ea69d865b38270ad), which mentioned the issue of exceeding the instructions limit. I'm sorry for not reading it carefully. You are right, and I need to correct myself.